src/corehost/cli/corehost_context_contract.h

```diff
@@ -66,4 +67,15 @@ static_assert(offsetof(corehost_context_contract, run_app) == 5 * sizeof(size_t)
 static_assert(offsetof(corehost_context_contract, get_runtime_delegate) == 6 * sizeof(size_t), "Struct offset breaks backwards compatibility");
 #pragma pack(pop)
 
+/// Signature of delegate returned by coreclr_delegate_type::component_create_native_delegate
+typedef int (__cdecl *ComponentCreateNativeDelegate)(const pal::char_t * assemblyPathNative /* Fully qualified path to assembly */,
+                                                     const pal::char_t * typeNameNative     /* Assembly qualified type name */,
+                                                     const pal::char_t * methodNameNative   /* Public static method name compatible with delegateType */,
+                                                     const pal::char_t * delegateTypeNative /* Assembly qualified delegate type name or null */,
+                                                     int32_t             flags              /* Extensibility flags (currently unused) */,
```

Comment:
Please document this must be `0`. We should also throw if it is not `0` - uninitialized memory is less than ideal.

Reply:
If these are flags, the type should be `uint32_t`.

Reply:
> If these are flags, the type should be uint32_t.

The coreclr C# implementation is currently using `int flags`. 

I looked at changing to `uint flags`, but I saw the note "unsigned integers other than Byte are not CLS-compliant".  Seems best not to use unsigned as it is not supported by all languages.

> We should also throw if it is not 0

That code would belong in coreclr.